AppCenter/AppCenter/MSAppCenter.m

```diff
@@ -416,18 +435,22 @@ - (void)applyPipelineEnabledState:(BOOL)isEnabled {
 - (void)initializeChannelGroup {
 
   // Construct channel group.
-  self.channelGroup = [MSChannelGroupDefault new];
-  if (self.appSecret) {
-    [self.channelGroup setAppSecret:self.appSecret];
-    [self.channelGroup attachSenderWithInstallId:self.installId logUrl:self.logUrl];
+  self.oneCollectorChannelDelegate =
+      self.oneCollectorChannelDelegate ?: [[MSOneCollectorChannelDelegate alloc] initWithInstallId:self.installId];
+  if (!self.channelGroup) {
+    self.channelGroup = [[MSChannelGroupDefault alloc] initWithInstallId:self.installId logUrl:self.logUrl];
+    [self.channelGroup addDelegate:self.oneCollectorChannelDelegate];
   }
-  self.oneCollectorChannelDelegate = [[MSOneCollectorChannelDelegate alloc] initWithInstallId:self.installId];
-  [self.channelGroup addDelegate:self.oneCollectorChannelDelegate];
 
   // Initialize a channel unit for start service logs.
-  self.channelUnit = [self.channelGroup
-      addChannelUnitWithConfiguration:[[MSChannelUnitConfiguration alloc]
-                                          initDefaultConfigurationWithGroupId:[MSAppCenter groupId]]];
+  if (self.channelUnit) {
```

Comment:
Can revisit this if statement from l446 to l453 to : 
```
     self.channelUnit = self.channelUnit :? [self.channelGroup
        addChannelUnitWithConfiguration:[[MSChannelUnitConfiguration alloc]
                                            initDefaultConfigurationWithGroupId:[MSAppCenter groupId]]];
     [self.channelUnit setAppSecret:self.appSecret];
```

Reply:
OK